pyalgostrategypool/options_bear_call_ladder_v2.py

```diff
@@ -0,0 +1,91 @@
+        - generalised docs in detail here : https://algobulls.github.io/pyalgotrading/strategies/common_options_strategy/
+"""
+
+
+class StrategyOptionsBearCallLadder(StrategyOptionsBaseV2):
+    name = 'Options Bear Call Ladder Template v2'
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        # Parameters (currently set with default values, can be overridden from the Parameters Configuration Pane)
+        self._leg_one_transaction_type = self.strategy_parameters.get('LEG_ONE_TRANSACTION_TYPE', 1)  # BUY: 1 | SELL: 2
+        self._leg_one_tradingsymbol_suffix = self.strategy_parameters.get('LEG_ONE_TRADING_SYMBOL_SUFFIX', 1)  # CE: 1 | PE: 2
+        self._leg_one_strike_direction = self.strategy_parameters.get('LEG_ONE_STRIKE_DIRECTION', 1)  # ITM: 0| ATM: 1| OTM: 2
+        self._leg_one_number_of_strikes = self.strategy_parameters.get('LEG_ONE_NUMBER_OF_STRIKES', 0)
+
+        self._leg_two_transaction_type = self.strategy_parameters.get('LEG_TWO_TRANSACTION_TYPE', 1)  # BUY: 1 | SELL: 2
+        self._leg_two_tradingsymbol_suffix = self.strategy_parameters.get('LEG_TWO_TRADING_SYMBOL_SUFFIX', 1)  # CE: 1 | PE: 2
+        self._leg_two_strike_direction = self.strategy_parameters.get('LEG_TWO_STRIKE_DIRECTION', 2)  # ITM: 0| ATM: 1| OTM: 2
+        self._leg_two_number_of_strikes = self.strategy_parameters.get('LEG_TWO_NUMBER_OF_STRIKES', 2)
+
+        self._leg_three_transaction_type = self.strategy_parameters.get('LEG_THREE_TRANSACTION_TYPE', 2)  # BUY: 1 | SELL: 2
+        self._leg_three_tradingsymbol_suffix = self.strategy_parameters.get('LEG_THREE_TRADING_SYMBOL_SUFFIX', 1)  # CE: 1 | PE: 2
+        self._leg_three_strike_direction = self.strategy_parameters.get('LEG_THREE_STRIKE_DIRECTION', 0)  # ITM: 0| ATM: 1| OTM: 2
+        self._leg_three_number_of_strikes = self.strategy_parameters.get('LEG_THREE_NUMBER_OF_STRIKES', 2)
+
+        # Maps
+        self.transaction_type_map = {1: "BUY", 2: "SELL"}
+        self.tradingsymbol_suffix_map = {1: "CE", 2: "PE"}
+        self.strike_direction_map = {0: "ITM", 1: "ATM", 2: "OTM"}
+
+        # Variables
+        self.number_of_allowed_expiry_dates = 1
+        self.instruments_done_for_the_day = None
+
+    def initialize(self):
+        super().initialize()
+        self.instruments_done_for_the_day = []
+
+    def get_child_instrument_details(self, base_instrument, tradingsymbol_suffix, strike_direction, no_of_strikes):
+        expiry_date = self.get_allowed_expiry_dates()[0]
+        child_instrument = self.get_options_instrument_with_strike_direction(base_instrument, expiry_date, tradingsymbol_suffix, strike_direction, no_of_strikes)
+        return child_instrument
+
+    def options_instruments_set_up_local(self, base_instrument, tradingsymbol_suffix, current_close, direction=None):
+        self.logger.info(f"delete this : {base_instrument} {tradingsymbol_suffix} {current_close} {direction}")
```

Comment:
`self.logger.info(f"delete this : {base_instrument} {tradingsymbol_suffix} {current_close} {direction}")`

have we reviewed line by line ?

Reply:
done